libkbfs/file_data.go

```diff
@@ -97,6 +122,341 @@ func (fd *fileData) getFileBlockAtOffset(ctx context.Context,
 	return ptr, parentBlocks, block, nextBlockStartOff, startOff, wasDirty, nil
 }
 
+// getLeafBlocksForOffsetRange fetches all the leaf ("direct") blocks
+// that encompass the given offset range (half-inclusive) in the file.
+// If `endOff` is -1, it returns blocks until reaching the end of the
+// file.  Note the range could be made up of holes, indicated in the
+// IPtrs list of the parent blocks.  If `prefixOk` is true, the
+// function will ignore context deadline errors and return whatever
+// prefix of the data it could fetch within the deadine.  Return
+// params:
+//
+//   * pathsFromRoot is a slice, ordered by file offset, of paths from
```

Comment:
Yeah I was thinking it might be useful for writes, though I haven't planned that part out yet completely and might just stick with the current serialized plan for writing for the moment.  It also made the recursion a little simpler.